libsel4/include/sel4/bootinfo_types.h

```diff
@@ -69,14 +71,36 @@ typedef struct seL4_BootInfo {
     seL4_SlotRegion   schedcontrol; /* Caps to sched_control for each node */
 #endif
     seL4_SlotRegion   untyped;         /* untyped-object caps (untyped caps) */
-    seL4_UntypedDesc  untypedList[CONFIG_MAX_NUM_BOOTINFO_UNTYPED_CAPS]; /* information about each untyped */
```

Comment:
It might also depend on the root cnode size.

Reply:
In `provide_cap()` there is a check about the space in the root C-Node and booting fails if it too small.

Reply:
That is a runtime check. if we would like to calculate the size during compilation, we should consider the factors and give one that does not fail.

Reply:
We could ensure at compile time that the root C-Node has at least `CONFIG_MAX_NUM_BOOTINFO_UNTYPED_CAPS` slots. But it's likely that it has far more slots anyway. Furthermore, in most cases it's even unlikely all slots are really used. And then the exact number of slots to be used is not know at compile time, as that depends on the memory layout - now one can assume this is static practically, but technically it depends on parameters passed from a previous boot loader. 
So not sure if there is really a gain here doing this as a compile time check also.

Reply:
I appreciate the effort, but I am not sure if the calculated one is much better than the user-defined one. I think several compile time assertions to ensure that a user does not provide a number that is greater than one-page size or root cnode size should be enough?

Reply:
I tend to agree, all this was motivated by https://github.com/seL4/seL4/issues/716 basically. Seems we could improve the comments here a bit why this exists and what changing this implies. The value is not really meant to be changed beyond a certain limit, that is driven by the free space in the boot info page on 64 bit platforms. So that makes we wonder if this should relly be a config option and not an internal constants.